### PR TITLE
Revert "Revert "record timing data for slow form submissions""

### PR DESCRIPTION
**Pull request by snopoke**
Reverts dimagi/commcare-hq#29439 which reverted #29395

Copied description from original PR:

## Summary
https://dimagi-dev.atlassian.net/browse/USH-836

The help diagnose performance issues during form submission processing this PR adds more granular timing data which get's reported to Sentry as breadcrumbs. In addition, if requests take longer than 60s an event is created in Sentry.

This extends the existing `LogLongRequestMiddleware` to allow configuring the timing threshold on a per view basis.

Can be reviewed by commit.

Note: this changes the sentry message:

```diff
- Request took a very long time.
+ Request timing above threshold
``` 

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Added tests

### QA Plan
Automated tests have been added that cover the changes to the functionality as well as the existing functionality.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by snopoke** [comment]
Updated with tests for reports and better error handling in the middleware. Also putting this on staging.

**Comment by snopoke** [comment]
Tested admin reports (and other reports / pages on staging)

**Review by biyeun — APPROVED** [review]
thank you @snopoke 🙌

File: corehq/middleware.py [hunk 1]
```diff
@@ -284,3 +312,26 @@ def process_request(self, request):
...
+def get_view_func(view_fn, view_kwargs):
+    """Given a view_fn from the `process_view` middleware function return the actual
+    function or class that represents the view.
+
+    :returns: the view function or class or None if not able to determine the view class
+    """
+    if getattr(view_fn, 'is_hq_report', False):  # HQ report
+        dispatcher = view_fn.view_class
+        domain = view_kwargs.get("domain", None)
+        slug = view_kwargs.get("report_slug", None)
```
Comment (millerdev) [inline corehq/middleware.py:326]: nit
```suggestion
        domain = view_kwargs.get("domain")
        slug = view_kwargs.get("report_slug")
```

**Pull request merged**